slatedb/src/batch.rs

```diff
@@ -314,34 +327,31 @@ impl WriteBatch {
     }
 }
 
-/// Iterator over WriteBatch entries
-pub(crate) struct WriteBatchIterator<'a> {
-    iter: Peekable<Box<dyn Iterator<Item = (&'a SequencedKey, RowEntry)> + Send + Sync + 'a>>,
+/// Iterator over `WriteBatch` entries.
+pub(crate) struct WriteBatchIterator {
+    iter: Peekable<Box<dyn Iterator<Item = (SequencedKey, RowEntry)> + Send + Sync>>,
     ordering: IterationOrder,
 }
 
-impl<'a> WriteBatchIterator<'a> {
+impl WriteBatchIterator {
     pub(crate) fn new(
-        batch: &'a WriteBatch,
+        batch: WriteBatch,
         range: impl RangeBounds<Bytes>,
         ordering: IterationOrder,
     ) -> Self {
         let range = KVTableInternalKeyRange::from(range);
-        let iter: Box<dyn Iterator<Item = _> + Send + Sync + 'a> = match ordering {
-            IterationOrder::Ascending => Box::new(
-                batch
-                    .ops
-                    .range(range)
-                    .map(|(k, v)| (k, v.to_row_entry(u64::MAX, None, None))),
-            ),
-            IterationOrder::Descending => Box::new(
-                batch
-                    .ops
-                    .range(range)
-                    .rev()
-                    .map(|(k, v)| (k, v.to_row_entry(u64::MAX, None, None))),
-            ),
-        };
+        let mut entries: Vec<(SequencedKey, RowEntry)> = batch
+            .ops
+            .range(range)
+            .map(|(k, v)| (k.clone(), v.to_row_entry(u64::MAX, None, None)))
+            .collect();
+
+        if matches!(ordering, IterationOrder::Descending) {
```

Comment:
are all of these ordering stuff to prepare for having reverse iterator exposed as an API ?

Reply:
I'm wondering the same. What's the motivation for this change vs. keeping the `match {}` we had before?